### ISSUE TITLE
[Bug] Browser extension takes a long time to get results

### ISSUE THREAD

**Issue by molant** [webhintio/hint#2316]
# 🐛 Bug report

## Description

Scan on https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Debugging takes a long time in both Firefox and Chrome (remember to disable all other extensions!)

## Environment

Using  the bundle in https://github.com/webhintio/hint/pull/2287

Disabling accessibility doesn't have any huge impact.

**Comment by molant** [comment]
After some investigation there are 2 causes:

* axe: there have been improvements in the latest weeks but I don't believe the changes have been published yet into stable. We should try with one of the beta versions they have see if this improves
* compat api: @antross is working on a refactor that should speed things up quite a bit

**Comment by molant** [comment]
Some tests using 3.2.2-canary.55820c on Firefox:

| website | 3.2.2 | 3.2.2.beta | Diff |
|---------|------|-----------|-----|
| edge insiders | 0:30 | 0:20 |  30% |
| youtube.com | Timeout | 2:15 | ♾|
| bing.com | 0:16 | 0:12 | 19% |

**Comment by molant** [comment]
I've tried running `axe` with the [same configuration as lighthouse](https://github.com/GoogleChrome/lighthouse/blob/fa053a92738dd867fc683ea5c0469d010fccf96d/lighthouse-core/gather/gatherers/accessibility.js#L25) but I don't see any difference with youtube (2:15) 😞

```js
let axeConfig = {
    elementRef: true,
    resultTypes: ['violations', 'inapplicable'],
    rules: {
        accesskeys: { enabled: true },
        'area-alt': { enabled: false },
        'aria-dpub-role-fallback': { enabled: false },
        'aria-hidden-body': { enabled: false },
        'aria-hidden-focus': { enabled: false },
        blink: { enabled: false },
        'duplicate-id-active': { enabled: false },
        'duplicate-id-aria': { enabled: false },
        'form-field-multiple-labels': { enabled: false },
        'html-xml-lang-mismatch': { enabled: false },
        marquee: { enabled: false },
        'server-side-image-map': { enabled: false },
        tabindex: { enabled: true },
        'table-fake-caption': { enabled: false },
        'td-has-header': { enabled: false }
    },
    runOnly: {
        type: 'tag',
        values: [
            'wcag2a',
            'wcag2aa'
        ]
    }
};
```

**Comment by antross** [comment]
The hint-compat-api refactor significantly improved performance for that hint. While not as big of a contributor to the overall analysis time as axe, the refactor reduced time analyzing youtube.com for the "Compatibility" section from ~40s to ~12s.

**Comment by antross** [comment]
Between the `hint-compat-api` refactor and `axe-core` updates this has improved dramatically. Closing as this no longer appears to be a significant issue.

Results from a recent test using `v0.0.6` of the webhint extension in Firefox:

| website | scan time |
|---------|-----------|
| [edge insiders]( https://www.microsoftedgeinsider.com/en-us/) | 6s |
| [youtube](https://youtube.com) | 45s (20s without Accessibility) |
| [bing](https://www.bing.com) | 5s |
